crates/ruff/src/commands/rule.rs

```diff
@@ -43,12 +44,16 @@ impl<'a> Explanation<'a> {
 
 fn format_rule_text(rule: Rule) -> String {
     let mut output = String::new();
-    output.push_str(&format!("# {} ({})", rule.as_ref(), rule.noqa_code()));
+    let _ = write!(&mut output, "# {} ({})", rule.as_ref(), rule.noqa_code());
     output.push('\n');
     output.push('\n');
 
     let (linter, _) = Linter::parse_code(&rule.noqa_code().to_string()).unwrap();
-    output.push_str(&format!("Derived from the **{}** linter.", linter.name()));
+    let _ = write!(
+        &mut output,
+        "Derived from the **{}** linter.",
+        linter.name()
+    );
```

Comment:
I kinda wonder if we should propose add something to `std` to make this nicer. I like the way the `x.push_str(&format!("..."))` construction reads much nicer, but I guess this approach avoids an intermediate allocation. But this approach has the very ugly `let _ = ...;` construction.

Reply:
That would be great. I definitely don't love it and it has the downside that later refactoring the code to eg use a Write instead of a string now incorrectly suppresses errors (instead of Rust telling you that this might now fail)